### ISSUE TITLE
[Android] UserLocationView shadow (and BearingTracking)

### ISSUE THREAD

**Issue by mayardb** [mapbox/mapbox-gl-native#4136]
Hi MapBox Team!

we have some few needs about the UserLocationView that we want to share with you:
- we need the UserLocationView to have a shadow to look like elevated
- we need the BearingTracking not to me disabled when starting a Gesture.

That's why we worked on these issues (Pull Request following).

**Comment by tobrun** [comment]
@mayardb 
Great, looking forward to the PR's coming from this. 

> we need the BearingTracking not to me disabled when starting a Gesture.

I have currenly exposed the following method in `TrackingSettings.java`

```java  
  public void setDismissTrackingOnGesture(boolean dismissTrackingOnGesture) {
        this.dismissTrackingOnGesture = dismissTrackingOnGesture;
        validateGesturesForTrackingModes();
    }
```
If false, this will not disabled gestures anymore in the way it used to. 
If this covering your use case? or do you need to explicitly stop only location tracking and not bearingtracking?

**Comment by tobrun** [comment]
I just noticed [this](https://github.com/mapbox/mapbox-gl-native/pull/4137/files#diff-886ea28eab2480af0b2b70f0b5c78898R2195) line in the PR where you are indeed managing them separately.

**Comment by tobrun** [comment]
@bleege @zugaldia:

The contents of the related PR contains following functionality:
 - managing tracking modes separately when a gesture occurs, this is now managed by one flag for both location and bearing tracking  with `setDismissTrackingOnGesture()`
 - expose API to change the UserLocationView drawables 
 - expose an API to add shadow to UserLocationView

**Comment by mayardb** [comment]
@tobrun:

> Is `setDismissTrackingOnGesture` covering your use case ?

It is not. We currently need the tracking to be dismissed when starting a gesture, but the bearing to stay. We also need the map not to head to user direction but this functionality is not managed in this Pull Request.

**Comment by tobrun** [comment]
This features has landed in #4396
